02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,71 @@
+const { people } = require('../data')
+
+
+const getPeople = (req, res) => {
+    res.status(200).json({ success: true, data: people });
+}
+
+
+const createPerson = (req, res) => {
+    const { name } = req.body
+    if (!name) {
+        return res.status(400)
+            .json({ success: false, msg: "Please provide a name" })
+    }
+    // adds the person to the json list
+    people.push({
+        id: people.length + 1, name: name
+    })
+
+    res.status(201).send({ success: true, person: name })
+}
+
+
+const createPersonPostman = (req, res) => {
+    const { name } = req.body
+    if (!name) {
+        return res.status(400)
+            .json({ success: false, msg: "Please provide a name" })
+    }
+    res.status(201).send({ success: true, data: [...people, name] })
+
+}
+
+const updatePerson = (req, res) => {
+    const { id } = req.params
+    const { name } = req.body
+    const person = people.find((person) => person.id === Number(id))
+
+    if (!name) {
+        return res
+            .status(404)
+            .json({ success: false, msg: `no person with id ${id}` })
```

Comment:
technically we want to keep error messages matched to the error they're catching (i.e. this should be for name, not id). Not a blocker, just a comment